src/mlpack/methods/ne/neat.hpp

```diff
@@ -713,32 +717,19 @@ class NEAT {
     }
 
     // Keep the best in each species.
-
-    //DEBUGGING!!!!!!!!!
-    printf("before, species sizes are: ");
-    for (ssize_t s=0; s<aPopulation.aSpecies.size(); ++s) {
-      std::cout<< aPopulation.aSpecies[s].aGenomes.size() << "  ";
-    }
-    printf("\n");
-    //DEBUGGING!!!!!!!!!
-
     CullSpeciesToOne(aPopulation);  // NOTICE: I found that this makes XOR test converges fast.
 
-    //DEBUGGING!!!!!!!!!
-    printf("after, species sizes are: ");
-    for (ssize_t s=0; s<aPopulation.aSpecies.size(); ++s) {
-      std::cout<< aPopulation.aSpecies[s].aGenomes.size() << "  ";
-    }
-    printf("\n");
-    //DEBUGGING!!!!!!!!!
-
     // Random choose species and breed child until reach population size.
     childGenomes.push_back(lastBestGenome);
```

Comment:
childGenomes.push_back(lastBestGenome);
Yeah, I use this to make sure the fitness will always decrease (at least not increase) during each iteration.
"it's possible that at the time we add the genome it doesn't exist anymore", sorry, but I am not quite clear about the meaning of this.